test/e2e/e2e_test.go

```diff
@@ -131,6 +133,159 @@ var _ = ginkgo.Describe("JobSet", func() {
 		})
 	})
 
+	ginkgo.When("job is unsuspended and suspend", func() {
+		ginkgo.It("should not create Jobs while suspended, and delete Jobs on suspend", func() {
+			ctx := context.Background()
+			js := shortSleepTestJobSet(ns).Obj()
+			jsKey := types.NamespacedName{Name: js.Name, Namespace: js.Namespace}
+
+			ginkgo.By("Create a suspended JobSet", func() {
+				js.Spec.Suspend = ptr.To(true)
+				js.Spec.TTLSecondsAfterFinished = ptr.To[int32](5)
+				gomega.Expect(k8sClient.Create(ctx, js)).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Verify Jobs aren't created", func() {
+				gomega.Consistently(func() int32 {
+					gomega.Expect(k8sClient.Get(ctx, jsKey, js)).Should(gomega.Succeed())
+					if js.Status.ReplicatedJobsStatus == nil {
+						return 0
+					}
+					return js.Status.ReplicatedJobsStatus[0].Active
+				}).WithTimeout(time.Second).WithPolling(200 * time.Millisecond).Should(gomega.Equal(int32(0)))
+			})
+
+			ginkgo.By("Unsuspend the JobSet setting schedulingGates that prevent pods from being scheduled", func() {
```

Comment:
Reading this by, I read this to mean that when you set schedulingGates we are unsuspended the job.

Reply:
But then you are also setting suspend to false in below not not exactly sure..

Reply:
SchedulingGates are for Pods. The Jobs are unsuspended. This is just to show scheduling gates can be added on resume. Also, I want to make sure the Pods don't actually run, so that the Job does not complete on the first resume.